api/python/cellxgene_census/src/cellxgene_census/experimental/_embedding.py

```diff
@@ -91,48 +149,119 @@ def get_embedding(
-            measurement_name = embedding_metadata["measurement_name"]
-            if measurement_name not in census["census_data"][experiment_name].ms:
-                raise ValueError("Census and embedding mismatch - measurement_name does not exist")
+    Returns:
+        A dictionary containing metadata describing the embedding.
 
-        embedding_shape = (len(obs_soma_joinids), E.shape[1])
-        embedding = np.full(embedding_shape, np.NaN, dtype=np.float32, order="C")
+    Raises:
+        ValueError: if no embeddings are found for the specified query parameters.
 
-        obs_indexer = soma.tiledbsoma_build_index(obs_soma_joinids, context=E.context)
-        for tbl in E.read(coords=(obs_soma_joinids,)).tables():
-            obs_idx = obs_indexer.get_indexer(tbl.column("soma_dim_0").to_numpy())
-            feat_idx = tbl.column("soma_dim_1").to_numpy()
-            emb = tbl.column("soma_data")
+    """
+    response = requests.get(CELL_CENSUS_EMBEDDINGS_MANIFEST_URL)
+    response.raise_for_status()
 
-            indices = obs_idx * E.shape[1] + feat_idx
-            np.put(embedding.reshape(-1), indices, emb)
+    manifest = cast(Dict[str, Dict[str, Any]], response.json())
+    embeddings = []
+    for _, obj in manifest.items():
+        if (
+            obj["embedding_name"] == embedding_name
+            and obj["experiment_name"] == organism
+            and obj["data_type"] == embedding_type
+            and obj["census_version"] == census_version
+        ):
+            embeddings.append(obj)
 
-    return embedding
+    if len(embeddings) == 0:
+        raise ValueError(f"No embeddings found for {embedding_name}, {organism}, {census_version}, {embedding_type}")
+
+    return sorted(embeddings, key=lambda x: x["submission_date"])[-1]
+
+
+def get_all_available_embeddings(census_version: str) -> list[dict[str, Any]]:
+    """Return a dictionary of all available embeddings for a given Census version.
+
+    Args:
+        census_version:
+            The Census version tag, e.g., ``"2023-12-15"``.
+
+    Returns:
+        A list of dictionaries, each containing metadata describing an available embedding.
+
+    Examples:
+        >>> get_all_available_embeddings('2023-12-15')
+        [{
+            'experiment_name': 'experiment_1',
+            'measurement_name': 'RNA',
+            'organism': "homo_sapiens",
+            'census_version': '2023-12-15',
+            'n_embeddings': 1000,
+            'n_features': 200,
+            'uri': 's3://bucket/embedding_1'
+        }]
+
+    """
+    response = requests.get(CELL_CENSUS_EMBEDDINGS_MANIFEST_URL)
+    response.raise_for_status()
+
+    embeddings = []
+    manifest = response.json()
+    for _, obj in manifest.items():
+        if obj["census_version"] == census_version:
+            embeddings.append(obj)
+
+    return embeddings
+
+
+def get_all_census_versions_with_embedding(
```

Comment:
What is the rationale for adding these simple "filtered" variants of "get_all_embeddings"?  Personally, I think one simple "get all" function that the user can then filter easily (e.g.  a list of dicts or whatever) is preferable as it covers more use cases, and is simpler to maintain.

Reply:
These methods were requested by @pablo-gar . I'll defer the decision to him.

Reply:
Yes, a convenience helper API goes a long way in this space